internal/gcsx/temp_file.go

```diff
@@ -227,8 +226,14 @@ func (tf *tempFile) CheckInvariants() {
 
 func (tf *tempFile) Destroy() {
 	tf.state = fileDestroyed
-	// Throw away the file.
+	// Throw away the file (for anonymous files).
 	tf.f.Close()
+
+	// TODO ezl: move this logic out once we refactor cache disk file handling outside of gcsx.Tempfile
+	// Throw away the cache file and metadata
+	os.Remove(fmt.Sprintf("%s.json", tf.f.Name()))
```

Comment:
When `local-file-cache` is true, we should not evict any cache entries (not for milestone 1/2). 

When `local-file-cache` is false, we should not have `json` files.

So, I'm confused when this would be needed.

Reply:
This is to maintain cache integrity.  The cache design already allows for dynamic writes and cache eviction when files become stale (out of sync with gcs).